### PR TITLE
Master to PostgreSQL version 14

### PR DESCRIPTION
**Pull request by jrgemignani**
** **PLEASE DO NOT MERGE** ** until thoroughly reviewed and approved by all reviewers. Reviewers must state their approval as a comment in the thread below. Additionally, this merge needs to be appropriately squashed before merging.

PR for moving the master branch up to PostgreSQL version 14.

It contains all of the latest work in the master branch in addition to all of the changes made to go from PostgreSQL version 13 to 14. When this PR is merged the master will be the latest and on PostgreSQL version 14.

Everyone listed as a reviewer needs to review this PR

### CODE REVIEW

**Comment by dehowef** [comment]
Looks good.

File: src/backend/executor/cypher_utils.c [hunk 1]
```diff
@@ -254,7 +254,7 @@ HeapTuple insert_entity_tuple_cid(ResultRelInfo *resultRelInfo,
     // Insert index entries for the tuple
     if (resultRelInfo->ri_NumIndices > 0)
     {
-        ExecInsertIndexTuples(elemTupleSlot, estate, false, NULL, NIL);
+        ExecInsertIndexTuples(resultRelInfo, elemTupleSlot, estate, false, false, NULL, NIL);
```
Comment (MuhammadTahaNaveed) [inline src/backend/executor/cypher_utils.c:257]: [formatting] exceeds 80

**Review by MuhammadTahaNaveed — APPROVED** [review]
Looks good. Just added some formatting suggestions.

**Comment by eyab** [comment]
Looks good. Great work!!

**Review by rafsun42 — APPROVED** [review]
Looks good

**Pull request merged**